crates/api/src/transport.rs

```diff
@@ -321,14 +338,21 @@ impl Transport for DefaultTransport {
         data: bytes::Bytes,
     ) -> BoxFut<'_, K2Result<()>> {
         Box::pin(async move {
+            let len = data.len() as u64;
             let enc = (K2Proto {
                 ty: K2WireType::Notify as i32,
                 data,
-                space: Some(space.into()),
+                space: Some(space.clone().into()),
                 module: None,
             })
             .encode()?;
-            self.imp.send(peer, enc).await
+            match self.imp.send(peer, enc).await {
+                Ok(()) => {
+                    self.tracker.track_sent(space, None, len);
+                    K2Result::Ok(())
+                }
+                Err(err) => K2Result::Err(err),
+            }
```

Comment:
I think that you could use the `?` operator here:

```suggestion
            self.imp.send(peer, enc).await?;
            self.tracker.track_sent(space, None, len);
            K2Result::Ok(())
```

Or even, a more functional approach using [inspect](https://doc.rust-lang.org/std/result/enum.Result.html#method.inspect):

```suggestion
            self.imp
                .send(peer, enc)
                .await
                .inspect(|_| self.tracker.track_sent(space, None, len))
```

This is just a preference/opinion though, so it's up to you :slightly_smiling_face: